packages/playwright-ct/src/fixture.ts

```diff
@@ -0,0 +1,615 @@
+          body: JSON.stringify({
+            event: "step:end",
+            id: payload.stepId,
+            test: testData,
+            error: payload.error ? parseError(payload.error) : null,
+          }),
+          contentType: REPLAY_CONTENT_TYPE,
+        });
+      }
+    };
+  }
+
+  // Set up page annotation hook
+  async function addAnnotation(event: string, id?: string, detail?: Record<string, any>) {
+    if (!id) {
+      return;
+    }
+
+    try {
+      await page.evaluate(ReplayAddAnnotation, [
+        event,
+        id,
+        JSON.stringify({ ...detail, test: testData }),
+      ] as const);
+    } catch (error) {
+      logError("ReplayCTFixture:FailedToAddAnnotation", { error });
+    }
+  }
+
+  // Set up client instrumentation listener
+  const csiListener: ClientInstrumentationListener = {
+    onApiCallBegin: ({ userData, params, apiName, frames }) => {
+      const step: TestStepInternal | undefined = userData;
+
+      if (!step?.stepId) {
+        return;
+      }
+
+      if (isReplayAnnotation(params)) {
+        ignoredSteps.add(step.stepId);
+        return;
+      }
+
+      if (expectSteps.has(step.stepId)) {
+        return;
+      }
+
+      if (!frames.length) {
+        ignoredSteps.add(step.stepId);
+        return;
+      }
+
+      testInfo.attach(`replay:step:start`, {
+        body: JSON.stringify({
+          event: "step:start",
+          id: step.stepId,
+          test: testData,
+          apiName,
+          category: step.category,
+          frames,
+          params: params ?? {},
+          title: step.title,
+          hook: testInfo._currentHookType?.(),
+        }),
+        contentType: REPLAY_CONTENT_TYPE,
+      });
+
+      return addAnnotation("step:start", step.stepId, {
+        apiName,
+        category: step.category,
+        title: step.title,
+        params: params || {},
+        frames,
+        hook: testInfo._currentHookType?.(),
+      });
+    },
+
+    onApiCallEnd: ({ userData, error, params }) => {
+      const step: TestStepInternal | undefined = userData;
+      if (!step?.stepId || ignoredSteps.has(step.stepId)) {
+        return;
+      }
+
+      testInfo.attach(`replay:step:end`, {
+        body: JSON.stringify({
+          event: "step:end",
+          id: step.stepId,
+          test: testData,
+          error: error ? parseError(error) : null,
+        }),
+        contentType: REPLAY_CONTENT_TYPE,
+      });
+
+      return addAnnotation("step:end", step.stepId, {
+        error: error ? parseError(error) : null,
+      });
+    },
+  };
+
+  const clientInstrumentation = playwright._instrumentation;
+  clientInstrumentation.addListener(csiListener);
+
+  // Enhanced mount function that tracks component lifecycle
+  const enhancedMount = async (component: any, options?: any) => {
+    console.log(
+      "[replay.io CT]: Enhanced mount called for component:",
+      component?.type?.name || component?.type || "Component"
+    );
```

Comment:
I think console.logs dont belong to this fixture. It should work transparently and shouldn't clobber up console with potentially unwanted console logs